### PR TITLE
nixos/mate: use upstream session

### PR DESCRIPTION
**Pull request by worldofpeace**
###### Motivation for this change
https://github.com/NixOS/nixpkgs/pull/79939

###### Things done
Tested in https://github.com/NixOS/nixpkgs/pull/79939.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jtojnar** [comment]
Same caveat as the Xfce sessions.

**Comment by worldofpeace** [comment]
I also noticed that the `XDG_CURRENT_DESKTOP` won't be right even with https://github.com/NixOS/nixpkgs/pull/79830. It's `MATE` in `mate.desktop`.

> Same caveat as the Xfce sessions.

:frowning: If only this was a `wayland-sessions`.

File: nixos/modules/services/x11/desktop-managers/mate.nix [hunk 1]
```diff
@@ -44,35 +44,32 @@ in
...
+          '') config.environment.systemPackages}
 
-        # Add mate-control-center paths to some XDG variables because its schemas are needed by mate-settings-daemon, and mate-settings-daemon is a dependency for mate-control-center (that is, they are mutually recursive)
-        ${addToXDGDirs pkgs.mate.mate-control-center}
+          # Add mate-control-center paths to some XDG variables because its schemas are needed by mate-settings-daemon, and mate-settings-daemon is a dependency for mate-control-center (that is, they are mutually recursive)
+          ${addToXDGDirs pkgs.mate.mate-control-center}
+      fi
+    '';
 
-        ${pkgs.mate.mate-session-manager}/bin/mate-session ${optionalString cfg.debug "--debug"} &
```
Comment (romildo) [inline nixos/modules/services/x11/desktop-managers/mate.nix]: With this change I suppose that  `services.xserver.desktopManager.mate.debug` is useless. Should it be removed too?

Comment (worldofpeace) [reply nixos/modules/services/x11/desktop-managers/mate.nix]: Ahh, no. I believe there's probably an environment variable you can set. Like gnome has `GNOME_SESSION_DEBUG`.

Comment (romildo) [reply nixos/modules/services/x11/desktop-managers/mate.nix]: > Ahh, no. I believe there's probably an environment variable you can set. Like gnome has `GNOME_SESSION_DEBUG`.

It does not have a similar environment variable.

I have submitted a PR to include it: https://github.com/mate-desktop/mate-session-manager/pull/230

Comment (worldofpeace) [reply nixos/modules/services/x11/desktop-managers/mate.nix]: Cool, added patch into the package.

**Review by romildo — APPROVED** [review]
LGTM. Tested on a VM.

**Comment by ilya-fedin** [comment]
Any news? :thinking:

**Pull request merged**